GavinJoyce/ember-headlessui
# 🐛 Listbox: select next/previous option when opening a listbox with no value

# ✨ What Changed & Why

Problem: In the case when a listbox has no default `@value`, opening the listbox and navigating with `ArrowUp`/`ArrowDown` does not function.

Solution: Fix `setNextOptionActive` and `setPreviousOptionActive` methods to work even when there is no active option.

# 👹 Before

![before](https://user-images.githubusercontent.com/70960245/132528261-73490dfb-56ce-46a1-9f34-645cb4bcf3fc.gif)


# 😇 After

![after](https://user-images.githubusercontent.com/70960245/132528815-505c39bc-1a74-4923-b41a-b41b29800e6b.gif)


# 🌈 Testing

1. Check out this branch
2. Change `tests/dummy/app/components/listboxes/basic.hbs:3` to have no default `@value`
2. Run `ember serve`
3. Navigate to http://localhost:4200/listbox/listbox-basic
4. Observe `ArrowUp`/`ArrowDown` opens the listbox and navigates up and down the list.
